### PR TITLE
Descriptive image error

### LINKED ISSUE #9351: "Input image exceeds pixel limit" error is too vague
### Astro Info

```block
Astro                    v4.0.2
Node                     v18.18.0
System                   macOS (arm64)
Package Manager          unknown
Output                   static
Adapter                  none
Integrations             @astrojs/mdx
                         @astrojs/sitemap
```

### If this issue only occurs in one browser, which browser is a problem?

_No response_

### Describe the Bug

When I start importing content into Astro I get the error:

```
 generating optimized images
16:06:30   ▶ /_astro/untitled-1.TT4nkGxv_Z2rFvbz.webp (before: 107kB, after: 15kB) (+30ms) (1/101)
16:06:30   ▶ /_astro/frame_1.G23pLodK_1l8Gb9.webp (before: 101kB, after: 54kB) (+48ms) (7/101)
node:internal/process/promises:288
            triggerUncaughtException(err, true /* fromPromise */);
            ^

[Error: Input image exceeds pixel limit]
```

Which image? I have many images. Astro probably knows which one, and it should tell me.

### What's the expected result?

The error message is instructive and identifies the file that caused the error

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-bvk5u3?file=src%2Fcontent%2Fblog%2Ffirst-post.md

### Participation

- [X] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by tmcw**
## Changes

- Fixes #9351 

Right now, if you have an image in your site that is too big, you get a message saying that an image was too big, but not which one. This PR improves the error so that it tells you which one.

## Testing

I really don't know how to test this, tbh - I'd probably use `npm link` but I don't know how that'd work with pnpm.

## Docs

This doesn't need docs: it's just an improved error.

### CODE REVIEW

**Comment by jacobthesheep** [comment]
I believe pnpm link is a thing

File: .changeset/famous-cars-sell.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'astro': minor
```
Comment (florian-lefebvre) [inline .changeset/famous-cars-sell.md:2]:
```suggestion
'astro': patch
```

**Review by florian-lefebvre — CHANGES_REQUESTED** [review]

**Comment by ematipico** [comment]
@tmcw how did you test your changes? I mean, how did you make sure you could see the new errors?

Regardless, here's a suggestion: run `pnpm build` inside the astro package , create a new astro project on your machine, inside your new astro project run `pnpm link ../astro/packages/astro` (fix the path, but you want to link the built and local version of your astro), then run your project in dev mode and try to recreate the errors.

Take screenshots to show us the results.

**Review by ematipico — APPROVED** [review]

File: packages/astro/src/core/errors/errors-data.ts [hunk 2]
```diff
@@ -706,6 +706,23 @@ export const MarkdownImageNotFound = {
 	hint: 'This is often caused by a typo in the image path. Please make sure the file exists, and is spelled correctly.',
 } satisfies ErrorData;
 
+/**
+ * @docs
+ * @see
+ * - [Images](https://docs.astro.build/en/guides/images/)
+ * @description
+ * Astro could not transform one of your image. Oftentimes, this is caused by a corrupted or malformed image and re-exporting the image from your image editor may fix this issue.
```
Comment (sarah11918) [inline packages/astro/src/core/errors/errors-data.ts:714]:
```suggestion
 * Astro could not transform one of your images. Often, this is caused by a corrupted or malformed image. Re-exporting the image from your image editor may fix this issue.
```

**Review by sarah11918 — APPROVED** [review]
Just tiny adjustments! Great error message!

**Pull request merged**